provider/core.rb

```diff
@@ -28,6 +28,112 @@ module Provider
     quiet: false
   }.freeze
 
+  # Responsible for generating a file
+  # with a given set of parameters.
+  class FileTemplate
+    include Compile::Core
+    attr_accessor :name
+    attr_accessor :object
+    attr_accessor :product
+    attr_accessor :version
+    attr_accessor :output_folder
+    attr_accessor :product_ns
+    attr_accessor :config
+    attr_accessor :manifest
+    attr_accessor :env
+
+    # Ansible stuff.
+    attr_accessor :example
+
+    # InSpec stuff.
+    attr_accessor :plural
+    attr_accessor :doc_generation
+    attr_accessor :attribute_file_name
+    attr_accessor :privileged
+    attr_accessor :property
+
+    # Terraform stuff.
+    attr_accessor :async
+    attr_accessor :resource_name
```

Comment:
This doesn't feel like a property of a File. Shouldn't this be a property on the Resource/Product instead?

Reply:
Agreed, these are weird. These are variables that we access in the templates. As such (for now), we should place them inside this class.

The `FileTemplate` class is meant to show all of the variables that the templates can access. If you've got a better idea for a name, let me know.